cmake/common.cmake

```diff
@@ -262,11 +262,13 @@ target_include_directories(cchost PRIVATE ${CCF_GENERATED_DIR})
 # Host is always built with verbose logging enabled, regardless of CMake option
 target_compile_definitions(cchost PRIVATE VERBOSE_LOGGING)
 
-if("sgx" IN_LIST COMPILE_TARGETS)
-  target_compile_definitions(cchost PUBLIC CCHOST_SUPPORTS_SGX)
-endif()
-if("virtual" IN_LIST COMPILE_TARGETS)
-  target_compile_definitions(cchost PUBLIC CCHOST_SUPPORTS_VIRTUAL)
+if(COMPILE_TARGET STREQUAL "sgx")
+  target_compile_definitions(cchost PUBLIC PLATFORM_SGX)
+elseif(COMPILE_TARGET STREQUAL "snp")
+  target_compile_definitions(cchost PUBLIC PLATFORM_SNP)
+  target_include_directories(cchost PRIVATE ${OE_INCLUDEDIR})
```

Comment:
Do we still need OE_INCLUDEDIR here?

Reply:
Probably not, but I wanted to start with a minimal change which maintains the current behaviour just with the new options